File: 124world/124world.js
/*
문제 출처 : https://programmers.co.kr/learn/courses/30/lessons/12899
 */

const solution = (n) => {
  let answer = '';
  let temp = n;

  while (temp > 0) {
    if (temp % 3 === 0) {
      answer = '4' + answer;
      temp = temp / 3 - 1;
    } else if (temp % 3 === 1) {
      answer = '1' + answer;
      temp = Math.floor(temp / 3);
    } else {
      answer = '2' + answer;
      temp = Math.floor(temp / 3);
    }
  }
  return answer;
};

export default solution;

File: Programmers/level2/124world/124_world.py
# 문제 출처 : https://programmers.co.kr/learn/courses/30/lessons/12899?language=python3

def solution(n):
    answer = ''
    while n > 0:
        if n % 3 == 0:
            answer = '4' + answer
            n = n // 3 - 1
        elif n % 3 == 1:
            answer = '1' + answer
            n = n // 3
        else:
            answer = '2' + answer
            n = n // 3

    print(answer)
    return answer
